spec/furniture/marketplace/stripe_events_controller_request_spec.rb

```diff
@@ -81,6 +82,113 @@
+
+      it "does not attempt to transfer the order to seller's Square dashboard" do
+        square_order_double = instance_double(Marketplace::SquareOrder, send_order: true)
+        call
+        expect(square_order_double).not_to(have_received(:send_order))
+      end
+    end
+
+    context "when Square notifications are enabled" do
+      before do
+        # Pins clock to an arbitrary time
+        travel_to Time.zone.local(1994)
+      end
+
+      let(:marketplace) { create(:marketplace, :with_square) }
+      let(:person) { create(:person) }
+      let(:shopper) { create(:marketplace_shopper, person: person) }
+      let(:order) { create(:marketplace_order, :with_products, status: :pre_checkout, marketplace: marketplace, shopper: shopper) }
+
+      # rubocop:disable RSpec/ExampleLength
+      it "sends the order to the seller's Square dashboard" do
+        allow(Marketplace::SquareIdempotencyKey).to receive(:generate).and_return("idemp_key_1234")
+
+        fake_square = instance_double(
+          Square::Client,
+          orders: instance_double(
+            Square::OrdersApi,
+            create_order: instance_double(
+              Square::ApiResponse,
+              body: Struct.new(:order).new({
+                id: "sq_order_id_1234"
+              })
+            )
+          ),
+          payments: instance_double(
+            Square::PaymentsApi,
+            create_payment: instance_double(
+              Square::ApiResponse,
+              body: Struct.new(:payment).new({
+                id: "sq_payment_id_1234"
+              })
+            )
+          )
+        )
+
+        allow(Square::Client).to receive(:new).and_return(fake_square)
+
+        call
+
+        expect(fake_square.orders).to have_received(:create_order).with({
+          body: {
+            idempotency_key: "idemp_key_1234",
+            order: {
+              customer_id: shopper.id,
+              fulfillments: [{
+                delivery_details: {
+                  deliver_at: "1994-01-01T00:00:00+00:00",
+                  recipient: {
+                    address: {
+                      address_line_1: nil
+                    },
+                    display_name: shopper.person.display_name,
+                    phone_number: nil
+                  },
+                  schedule_type: "SCHEDULED"
+                },
+                state: "PROPOSED",
+                type: "DELIVERY"
+              }],
+              line_items: [{
+                base_price_money: {
+                  amount: order.ordered_products.first.product.price.cents,
+                  currency: "USD"
+                },
+                item_type: "ITEM",
+                name: order.ordered_products.first.name,
+                quantity: "1"
+              }],
+              location_id: ENV.fetch("MARKETPLACE_VENDOR_SQUARE_LOCATION", "square_location_id_1234"),
```

Comment:
Instead of coupling to `ENV`, can we get this value from the `marketplace`?

Reply:
Of course!